src/app/(main)/product/page.tsx

```diff
@@ -0,0 +1,180 @@
+'use client';
+
+import React from 'react';
+import ProductList from '@cps/components/product/productList';
+import { useAllProducts } from '@cps/trpc/use-product';
+import DOMPurify from 'dompurify';
+import Button from '@cps/components/button';
+
+const ProductsPage: React.FC = () => {
+  const { data: rawProducts, isLoading, error } = useAllProducts();
+
+  if (isLoading) {
+    return <div className="text-center py-10">Loading products...</div>;
+  }
+
+  if (error) {
+    return <div className="text-center py-10 text-red-500">Error: {error.message}</div>;
+  }
+
+  const products = rawProducts?.map((product) => ({
+    id: product.id,
+    title: DOMPurify.sanitize(product.name),
+    imageUrl: product.images?.[0]?.url || '/placeholder.png', // Fallback if no image
+    price: product.prices?.[0]
+  ? `${(product.prices[0].price / 100).toFixed(1)} ${product.prices[0].currency}`
+  : 'Price not available',
+    stock: product.stocks?.[0]?.stock || 'Stock not available',
+  }));
+  
+
+  return (
+<div className="grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4 min-h-screen">
+
+      {/* Main Content */}
+      <nav className="bg-neutral p-4 flex justify-between">
+        <div className="absolute left-1/2 transform -translate-x-1/2">
+          <h1 className="text-lg font-bold">GPS Demo</h1>
+        </div>
+        <div className="flex items-center gap-4">
+          <a href="#" className="text-sm hover:underline">Home</a>
+          <a href="#" className="text-sm hover:underline">Store</a>
+        </div>
+        <div className="flex gap-2">
+          <Button className="w-8 h-8 rounded-full bg-neutral-light flex justify-center items-center">T</Button>
+          <Button className="w-8 h-8 rounded-full bg-neutral-light flex justify-center items-center">C</Button>
+          <Button className="w-8 h-8 rounded-full bg-neutral-light flex justify-center items-center">A</Button>
+        </div>
+      </nav>
```

Comment:
This is a menu. You typically want to use it in a `layout.tsx` component.

Reply:
This can be done in a another component, and imported again in a `layout.tsx` again